### DISCUSSION TITLE
`nullable` generic type constraint

### DISCUSSION THREAD

**Discussion by RenderMichael** [dotnet/csharplang#6931] [Language Ideas]
The generic constaint `where T : nullable` would be useful because it would allow
``` csharp
T myT = null;
```
instead of `default`, which may not be null for non-nullable value types.

This would need special handling for the value-type vs the reference-type generic instantiations, which would boil down to  `where T : Nullable<> or class?`

**Comment by Joe4evr** [comment]
This has been asked for many times before, and the short answer is that it's impossible to implement for the simple reason that there's no IL sequence that can represent `T` for a reference type (because those are still the same type at runtime) and `Nullable<T>` for a value type (because it's a seperate type for value types).
